dashboard/utils/fine_tuning/openai.ts

```diff
@@ -0,0 +1,116 @@
+import { JsExposedEnv } from "../minijinja/pkg";
+import {
+  ParsedInferenceRow,
+  ContentBlockOutput,
+  JsonInferenceOutput,
+} from "utils/clickhouse";
+import { render_message } from "./rendering";
+import OpenAI from "openai";
+import { createReadStream } from "fs";
+import * as fs from "fs/promises";
+import * as path from "path";
+
+type OpenAIRole = "system" | "user" | "assistant";
+
+type OpenAIMessage = {
+  role: OpenAIRole;
+  content: string;
+};
+
+const client = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+export function sample_to_openai_messages(
+  sample: ParsedInferenceRow,
+  env: JsExposedEnv,
+) {
+  const system = sample.input.system;
+  const messages: OpenAIMessage[] = [];
+  if (env.has_template("system")) {
+    const rendered_system = env.render("system", system);
+    messages.push({
+      role: "system",
+      content: rendered_system,
+    });
+  } else if (system) {
+    if (typeof system !== "string") {
+      throw new Error(
+        "System message must be a string when not using templates",
+      );
+    }
+    messages.push({
+      role: "system",
+      content: system,
+    });
+  }
+  for (const message of sample.input.messages) {
+    const rendered_message = render_message(env, message.role, message.content);
+    messages.push({
+      role: message.role,
+      content: rendered_message,
+    });
+  }
+  // Add type check at the beginning
+  const isChatInference = Array.isArray(sample.output);
```

Comment:
This is not the right way to do this

Reply:
I think the right way to do it is actually pass the function type in. We can add this as we actually wire the whole thing up.

Reply:
ok